File: Plugins/Aspose_Barcode_Java_for_Ruby/lib/asposebarcodejava/BarcodeRecognition/barcodeorientation.rb
module Asposebarcodejava
  module BarcodeOrientation
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # initialize barcode reader
        img = data_dir + "orientation.jpg"
        barcode_reader_type = Rjb::import('com.aspose.barcoderecognition.BarCodeReadType')
        reader = Rjb::import('com.aspose.barcoderecognition.BarCodeReader').new(img, barcode_reader_type.Code39Standard)
        
        # Set orientation
        reader.setOrientationHints(Rjb::import('com.aspose.barcoderecognition.RecognitionHints$Orientation').Rotate90)

        # Try to recognize all possible barcodes in the image
        while reader.read()
            # print the code text, if barcode found
            puts "CodeText: " + reader.getCodeText().to_s
        end

        # Close reader
        reader.close()
    end
  end
end

File: Examples/src/main/java/com/aspose/barcode/examples/barcode_recognition/advanced_features/BarcodeOrientation.java
package com.aspose.barcode.examples.barcode_recognition.advanced_features;

import com.aspose.barcode.barcoderecognition.BarCodeReader;
import com.aspose.barcode.barcoderecognition.BarCodeResult;
import com.aspose.barcode.barcoderecognition.DecodeType;
import com.aspose.barcode.examples.ApplyALicense;
import com.aspose.barcode.examples.Utils;

public class BarcodeOrientation {

	public static void main(String[] args) throws Exception {
		ApplyALicense.applyALicense();
		// ExStart: BarcodeOrientation
		// The path to the resource directory.
		String dataDir = Utils.getDataDir(BarcodeOrientation.class) + "BarcodeReader/advanced_features/";

		// Read code39 barcode from image
		String image = dataDir + "code39Extended.jpg";
		BarCodeReader reader = new BarCodeReader(image, DecodeType.CODE_39_STANDARD);

		// Barcode orientation is detected automatically

		// Try to recognize all possible barcodes in the image
		for (BarCodeResult result : reader.readBarCodes()) {
			System.out.println("BarCode CodeText: " + result.getCodeText());
			System.out.println("BarCode CodeType: " + result.getCodeTypeName());
		}
		// ExEnd: BarcodeOrientation
	}

}
